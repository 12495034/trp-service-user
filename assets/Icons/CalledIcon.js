import React from 'react'
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';


export default function CalledIcon(props) {
    if (props.checkedIn == true && props.called == false) {
        return <MaterialCommunityIcons name="bell-outline" color="black" size={25} />
    } else if (props.checkedIn == true && props.called == true) {
        return <MaterialCommunityIcons name="bell" color="purple" size={25} />
    } else {
        return null
    }

}
