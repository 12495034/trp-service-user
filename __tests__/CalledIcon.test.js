import React from 'react';
import renderer from 'react-test-renderer';
import CalledIcon from '../assets/Icons/CalledIcon';

test('check render of called Icon for checked in and called', () => {
  const tree = renderer.create(<CalledIcon 
    checkedIn={true} 
    called={true}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});


test('check render of called Icon for checked in and not called', () => {
    const tree = renderer.create(<CalledIcon 
      checkedIn={true} 
      called={false}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('check render of called Icon for not checked in and not called', () => {
    const tree = renderer.create(<CalledIcon 
      checkedIn={false} 
      called={false}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });