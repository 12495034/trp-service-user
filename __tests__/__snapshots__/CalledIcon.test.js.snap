// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render of called Icon for checked in and called 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    [
      {
        "color": "purple",
        "fontSize": 25,
      },
      undefined,
      {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      {},
    ]
  }
>
  󰂚
</Text>
`;

exports[`check render of called Icon for checked in and not called 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    [
      {
        "color": "black",
        "fontSize": 25,
      },
      undefined,
      {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      {},
    ]
  }
>
  󰂜
</Text>
`;

exports[`check render of called Icon for not checked in and not called 1`] = `null`;
